### PR TITLE
[Merged by Bors] - chore: adapt install scripts from mathlib-tools

### PR DESCRIPTION
**Pull request by kim-em**
This takes the installation scripts from `mathlib-tools` and adapts them for `lean4`.

I will use these in the new installation instructions to replace 

* https://leanprover-community.github.io/install/macos.html
* https://leanprover-community.github.io/install/debian.html

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: scripts/install_macos.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+set -exo pipefail
+
+# Install Homebrew
+set -e
```
Comment (jcommelin) [inline scripts/install_macos.sh:6]: This seems redundant now.

**Review by jcommelin — COMMENTED** [review]
Should we standardize the installation order of `code`, lean4 extension, `elan` in these scripts?

**Comment by jcommelin** [comment]
bors d+

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**